include/GafferSceneUI/ContextAlgo.h

```diff
@@ -74,6 +74,10 @@ void expand( Gaffer::Context *context, const GafferScene::PathMatcher &paths, bo
 /// Returns a new PathMatcher containing the new leafs of this expansion.
 GafferScene::PathMatcher expandDescendants( Gaffer::Context *context, const GafferScene::PathMatcher &paths, const GafferScene::ScenePlug *scene, int depth = Imath::limits<int>::max() );
 
+/// Removes paths from the current expansion.
+/// Returns a new PathMatcher containing the removed paths.
+GafferScene::PathMatcher collapse( Gaffer::Context *context, const GafferScene::PathMatcher &paths );
```

Comment:
Do you need this one for something, and if so, how is it fitting? To me it feels like the details of the internal implementation are still really tied to the specific needs of `SceneView::collapseSelection()`, to the point where I'm not sure  this is the behaviour I'd expect from collapse.